autosklearn/data/feature_validator.py

```diff
@@ -76,23 +148,32 @@ def _fit(
 
         Parameters
         ----------
-            X: typing.Union[np.ndarray, pd.DataFrame]
+            X: SUPPORTED_FEAT_TYPES
                 A set of features that are going to be validated (type and dimensionality
                 checks) and a encoder fitted in the case the data needs encoding
         """
         if hasattr(X, "iloc") and not scipy.sparse.issparse(X):
+            X = typing.cast(pd.DataFrame, X)
             # Treat a column with all instances a NaN as numerical
+            # This will prevent doing encoding to a categorical column made completely
+            # out of nan values -- which will trigger a fail, as encoding is not supported
+            # with nan values.
+            # Columns that are completely made of NaN values are provided to the pipeline
+            # so that later stages decide how to handle them
             if np.any(pd.isnull(X)):
                 for column in X.columns:
                     if X[column].isna().all():
                         X[column] = pd.to_numeric(X[column])
 
-            self.enc_columns, self.feature_types = self._get_columns_to_encode(X)
+            self.enc_columns, self.feat_type = self._get_columns_to_encode(X)
 
             if len(self.enc_columns) > 0:
 
                 self.encoder = make_column_transformer(
-                    (preprocessing.OrdinalEncoder(), self.enc_columns),
+                    (preprocessing.OrdinalEncoder(
+                        handle_unknown='use_encoded_value',
+                        unknown_value=np.nan,
```

Comment:
I think -1 would be more appropriate here to avoid confusing this with real np.NaN values (in the future when we support them).